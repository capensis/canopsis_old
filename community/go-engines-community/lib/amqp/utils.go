package amqp

func BuildRoutingKey(prefix string, initiator string) string {
	return prefix + "_" + initiator
}
