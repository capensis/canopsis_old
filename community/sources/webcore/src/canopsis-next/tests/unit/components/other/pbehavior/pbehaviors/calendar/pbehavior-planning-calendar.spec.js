import { flushPromises, createVueInstance, generateRenderer } from '@unit/utils/vue';
import {
  createInfoModule,
  createMockedStoreModules,
  createPbehaviorTimespanModule,
  createPbehaviorTypesModule,
} from '@unit/utils/store';

import PbehaviorPlanningCalendar from '@/components/other/pbehavior/calendar/pbehavior-planning-calendar.vue';

const localVue = createVueInstance();

const stubs = {
  'c-action-btn': true,
  'c-progress-overlay': true,
  'c-calendar': {
    template: `
      <div class="c-calendar">
        <slot name="form-event" :close="close" :event="event" />
        <slot name="menu-right" />
      </div>
    `,
    computed: {
      event() {
        return {};
      },
    },
    methods: {
      close: jest.fn(),
    },
  },
  'pbehavior-create-event': true,
  'pbehavior-planning-calendar-legend': true,
};

describe('pbehavior-planning-calendar', () => {
  jest.useFakeTimers({ now: 1386435500000 });

  const { infoModule, shownUserTimezone } = createInfoModule();
  const { pbehaviorTimespanModule } = createPbehaviorTimespanModule();
  const { pbehaviorTypesModule } = createPbehaviorTypesModule();
  const store = createMockedStoreModules([infoModule, pbehaviorTimespanModule, pbehaviorTypesModule]);

  const snapshotFactory = generateRenderer(PbehaviorPlanningCalendar, {
    localVue,
    stubs,
    parentComponent: {
      provide: {
        $system: {},
      },
    },
  });

  test('Renders `pbehavior-planning-calendar` with required props', async () => {
    const wrapper = snapshotFactory({
      store,
      propsData: {
        pbehaviorsById: {},
        addedPbehaviorsById: {},
        removedPbehaviorsById: {},
        changedPbehaviorsById: {},
      },
    });

    await flushPromises();

    expect(wrapper).toMatchSnapshot();
  });

  test('Renders `pbehavior-planning-calendar` with required props and with shownUserTimezone', async () => {
    shownUserTimezone.mockReturnValueOnce(true);

    const wrapper = snapshotFactory({
      store: createMockedStoreModules([infoModule, pbehaviorTimespanModule, pbehaviorTypesModule]),
      propsData: {
        pbehaviorsById: {},
        addedPbehaviorsById: {},
        removedPbehaviorsById: {},
        changedPbehaviorsById: {},
      },
    });

    await flushPromises();

    expect(wrapper).toMatchSnapshot();
  });
});
