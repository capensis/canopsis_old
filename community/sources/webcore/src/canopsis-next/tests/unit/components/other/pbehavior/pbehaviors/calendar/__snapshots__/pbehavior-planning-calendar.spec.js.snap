// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pbehavior-planning-calendar Renders \`pbehavior-planning-calendar\` with required props 1`] = `
<div
  class="fill-height"
>
  <div
    class="c-calendar"
    color="primary"
    events=""
    timezone="Europe/Paris"
  >
    <pbehavior-create-event-stub
      defaultname=""
      event="[object Object]"
      timezone="Europe/Paris"
    />
    <pbehavior-planning-calendar-legend-stub
      exceptiontypes=""
    />
  </div>
</div>
`;
